stat/stat_test.go

```diff
@@ -1625,5 +1626,75 @@ func TestStdScore(t *testing.T) {
 			t.Errorf("StdScore mismatch case %d. Expected %v, Found %v", i, test.z, z)
 		}
 	}
+}
+
+func ExampleQuantile_median() {
+	var weights []float64
+	x1 := []float64{1, 3, 3, 6, 7, 8, 9}
+	// the median of a dataset is also known as the 2-quantile of a dataset.
+	m1 := Quantile(0.5, AvgDiscont, x1, weights)
+	fmt.Printf("median = %v\n", m1)
 
+	x2 := []float64{1, 2, 3, 4, 5, 6, 8, 9}
+	m2 := Quantile(0.5, AvgDiscont, x2, weights)
+	fmt.Printf("median = %v\n", m2)
+
+	// Output:
+	// median = 6
+	// median = 4.5
+}
+
+func TestQuantileMedian(t *testing.T) {
```

Comment:
I think this test should be replaced with a `TestQuantileHalfBetween` (or whatever we call it). We need a more robust set of tests for this case before it can go in. Tests like `xs := []float64{10,20}` should return `10` for `p = 0`, `20` for `p = 1`, and `15` for all other cases. Similarly, there should be weighted tests, where if the `p` matches the weights perfectly, the correct `x` is returned, otherwise it's the average of the two `xs`.